src/controllers/userController.js

```diff
@@ -59,8 +64,43 @@ const logout = async (req, res) => {
   }
 };
 
+const updateUser = async (req, res) => {
+  try {
+    const token = req.headers.authorization?.split(' ')[1];
+    if (!token) {
+      throw new UnauthenticatedError('Authentication token missing');
+    }
+
+    const decoded = jwt.verify(token, process.env.JWT_SECRET);
+    const userId = decoded.userId;
+
+    const { firstName, lastName, location } = req.body;
+    if (!firstName && !lastName && !location) {
+      throw new BadRequestError('No valid fields to update');
+    }
+
```

Comment:
Make sure that undefined doesnt bypass this if check, or they could get set to undefined in the DB.: 
if (firstName !== undefined) updates.firstName = firstName;
if (lastName !== undefined) updates.lastName = lastName;
if (location !== undefined) updates.location = location;

if (Object.keys(updates).length === 0) {
  throw new BadRequestError('No valid fields to update');
}

const updatedUser = await User.findByIdAndUpdate(
  userId,
  { $set: updates },
  { new: true, runValidators: true }
);